src/reducers/wallet.js

```diff
@@ -1,13 +1,28 @@
-import { WALLET_READ_SUCCESS, WALLET_CREATE_SUCCESS } from '../actions';
+import { WALLET_READ_REQUEST, WALLET_READ_SUCCESS, WALLET_READ_FAILURE, WALLET_CREATE_SUCCESS } from '../actions';
 
-const initialState = [];
+const initialState = { list: [] };
 
 const reducer = (state = initialState, { type, payload }) => {
   switch (type) {
+    case WALLET_READ_REQUEST:
+      return { ...state, isLoading: true };
```

Comment:
`WALLET_READ_REQUEST` is dispatched when the API call has fired. Since the `state` is immutable, we need to create another object. In older versions of JavaScript/ECMAScript (let's just refer to it as `ES` from now on), it was possible to do it like this:
`Object.assign({}, state, { isLoading: true })`
However, in newer ES you can use the following syntax sugar:
`{ ...obj1, ...obj2, hello: 'world' }`

The above will create a new object with keys merged from `obj1`, then from `obj2`, and finally, appending the `hello` key.

So, if our state is in it's initial form, then:
`{ list: [] }` + `{ isLoading: true }` => `{ list: [], isLoading: true}`